src/views/dialogs/TransferDialogs/WithdrawDialog2/withdrawHooks.ts

```diff
@@ -129,3 +141,44 @@ export function useWithdrawStep({
     isLoading,
   };
 }
+
+export function useProtocolWithdrawalValidation({
+  freeCollateral,
+  withdrawAmount,
+  selectedRoute,
+}: {
+  freeCollateral?: BigNumber;
+  withdrawAmount: string;
+  selectedRoute?: RouteResponse;
+}): string | undefined {
+  const stringGetter = useStringGetter();
+  const selectedLocale = useAppSelector(getSelectedLocale);
+  const { decimal: decimalSeparator, group: groupSeparator } = useLocaleSeparators();
+  const { usdcWithdrawalCapacity } = useWithdrawalInfo({ transferType: 'withdrawal' });
+  const withdrawAmountBN = MustBigNumber(withdrawAmount);
+
+  if (withdrawAmount === '' || withdrawAmountBN.lte(0) || !selectedRoute) {
+    return undefined;
+  }
+
+  if (freeCollateral && withdrawAmountBN.gt(freeCollateral)) {
```

Comment:
shouldnt this error also if freeCollateral is 0 and withdrawAmountBn > 0n

Reply:
or maybe not idk

Reply:
freeCollateral 0 and withdrawAmount > freeCollateral would error. This is the current free collateral not the post withdraw free collateral.